Documentation/contributing/development/dev_setup.rst

```diff
@@ -62,9 +62,9 @@ in order to effectively contribute to Cilium:
 +===================================================================+==============================+=================================================================+
 |  git                                                              | latest                       | N/A (OS-specific)                                               |
 +-------------------------------------------------------------------+------------------------------+-----------------------------------------------------------------+
-|  clang                                                            | >= 10.0 (latest recommended) | N/A (OS-specific)                                               |
+|  clang                                                            | >= 14.0 (latest recommended) | N/A (OS-specific)                                               |
```

Comment:
the build is running with clang 17, while clang 14+ is working now, it might break without us noticed, wdyt about mentioning clang 17 directly?

Reply:
The nice thing about LLVM 14 is that it's the default with Ubuntu 22.04 :sweat_smile: But I don't really have an opinion on this. I'm unsure what approach we followed for other lines in the table :thinking:

Reply:
oh i see, I am good with clang  14+ as well.

(Ubuntu 24.04 is with clang 18 by default :scream: )

Reply:
Let's see what others thing.

Reply:
Same feedback from Daniel so I just updated it.